### PR TITLE
Dynamically get all teachers for assigning to a class

### LINKED ISSUE #79: Add query parameter to users to filter output
Addas a query parameter to the users APi to filter the ouput by role.

### LINKED ISSUE #80: Update CreateEvent component to get a list of all teachers and add them via dropdown
Title. Use Material UI and the UsersAPI to get a master list of teachers and render the dropdown with all the teachers in the org,

### PR DESCRIPTION
**Pull request by Anshul-Birla**
### Administrative Info

Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Github Issue]-[3-4 word description separated by dashes]`.

### What issue(s) does this branch close?
Closes #79 
Closes #80 

### Changes

- Added a get route to /api/users that gets all users
- Added a way to filter by role when making get requests
- Added a way to dynamically get all teachers and let users select them 

How did you confirm your changes worked?

- Made multiple teacher accounts and see if making a class with them worked 
- Tested API route with tests

### Confirmation of Change

- Check above

### CODE REVIEW

**Review by zainnk902 — CHANGES_REQUESTED** [review]
Great work! Just had a small change with testing, but everything looks really good.

 One question I had that might just be a design choice but thought I'd mention it; when someone wants to create a class, the dropdown for the teachers available when there are no teachers is just an empty dropdown. There shouldn't be too many cases when there are no teachers, but maybe instead of having an empty dropdown, it just has some error message like "No teachers available"? 

Just a thought, but everything else looked fine.

**Comment by Anshul-Birla** [comment]
> Great work! Just had a small change with testing, but everything looks really good.
> 
> One question I had that might just be a design choice but thought I'd mention it; when someone wants to create a class, the dropdown for the teachers available when there are no teachers is just an empty dropdown. There shouldn't be too many cases when there are no teachers, but maybe instead of having an empty dropdown, it just has some error message like "No teachers available"?
> 
> Just a thought, but everything else looked fine.

I added some more tests :). 
In the real app, there are not going to be any cases where there are no teachers so i don't think its super necessary for the create class modal to have a no teachers view :). Thanks for the idea though :)

File: pages/api/users/index.ts [hunk 1]
```diff
@@ -30,6 +30,18 @@ const userHandler: NextApiHandler = async (req: NextApiRequest, res: NextApiResp
       }
     }
 
+    case "GET": {
+      let role: Roles | undefined = undefined;
+      if (req.query && req.query.role) role = req.query.role as Roles;
```
Comment (allisonbhavsar) [inline pages/api/users/index.ts:35]: Should the GET route return 400 bad request error if the "role" query param is invalid (i.e. not one of Admin/Volunteer/Student/Staff/Parent)? Currently it returns an empty array on an invalid query param with status code 200

File: __tests__/users.test.ts [hunk 2]
```diff
@@ -14,19 +14,215 @@ beforeAll(async () => {
...
+      "/api/users/",
+      query,
+      "GET",
+      undefined,
+      StatusCodes.OK,
+      expected
+    );
+  });
+});
+
```
Comment (allisonbhavsar) [inline __tests__/users.test.ts:225]: Add a test case where the value of the role query param is an invalid Role? Although I'm not sure what you want the expected behavior to be in this case, I made another comment about this in pages/api/users/index.ts

**Review by allisonbhavsar — CHANGES_REQUESTED** [review]
Looks good, just left a couple of comments about the role query param in the GET route for all users

**Review by zainnk902 — APPROVED** [review]
lgtm 🤝

**Review by allisonbhavsar — APPROVED** [review]
LGTM

**Pull request merged**